### ISSUE TITLE
Surface Pen Input Bug

### ISSUE THREAD

**Issue by Dariqq** [xournalpp/xournalpp#3124]
**Affects versions :**
 - OS: Windows 10
 - Which version of libgtk do you use: 3.24.23
 - Version of Xournal++: 1.0.20
 - Installation method: official windows installer

**Describe the bug**

I am using a Surface Go with the Surface pen, which has an eraser on the other side of the pen. At first switching between the pen and the eraser everything works as intended. But when I try to erase something with the back after the first time, the eraser still gets highlighted in the toolbar and the cursor has the eraser shape but nothing is being erased.

**To Reproduce**

1. Write lines with the pen
2. Turn pen around to switch to eraser (erases lines)
3. Write some lines again
4. Turn pen around to switch again (gets recognised as eraser in the toolbar, but does not erase)

**Expected behavior**

Turning the pen to switch to the eraser should always erase lines.

**Screenshots of Problem**

Screenshot of input device settings
<img width="695" alt="Screenshot_2021-05-22_234403" src="https://user-images.githubusercontent.com/77271900/119266248-476d0300-bbea-11eb-9b66-5c050d15bcff.png">

**Additional context**

After installing the WinTab drivers (https://github.com/xournalpp/xournalpp/issues/1346#issuecomment-519718882) and with the experimental input system the pen gets  recognised as an input device. I am also also neither  able to use a mouse for drawing/ menu selecting after using the pen nor do the pen buttons work, which may be related.

The same issue occurs on the 1.1.0+dev-nightly release using libgtk 3.24.29

**Comment by rolandlo** [comment]
Input recognition on MS Windows will improve greatly after GTK+ adds support for Windows Pointer Input stack. See https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/1563 for a Merge Request in progress currently under review. It will most likely not take very long until it gets merged. See #2316 for the corresponding Xournal++ issue. 
If you would like to help testing, you could follow the recipe to build `libgdk-3-0.dll` and `libgtk-3-0.dll`, put them into your Xournal++ installation as described in [this answer](https://github.com/xournalpp/xournalpp/issues/2316#issuecomment-840529906) and see, if it solves your problems.

**Comment by Dariqq** [comment]
Thank you for the tip. Unfortunately I got errors when trying to compile with msys2. However using the test build in the comment below (https://github.com/xournalpp/xournalpp/issues/2316#issuecomment-840541487) I can verify that my issue is resolved there, meaning the switching between Eraser and Pen now works as it should. 

Some minor issues remain, like buttons on the pen getting recognized as being pressed and the selected action getting highlighted in the menu bar, but the button action is not being performed when drawing.

**Comment by rolandlo** [comment]
The patched GTK+ 3 has now been merged. You can copy the `libgdk-3-0.dll` and `libgtk-3-0.dll` into your [Xournal++ 1.1.0+dev](https://github.com/xournalpp/xournalpp/releases/tag/nightly) installation as described [here](https://github.com/xournalpp/xournalpp/issues/2316#issuecomment-873435907). This may solve the issue.

**Comment by rolandlo** [comment]
The issue should be fixed in [Version 1.1.0-hotfix+1](https://github.com/xournalpp/xournalpp/releases/tag/1.1.0-hotfix.1). Please report if you experience otherwise.

**Comment by digitalsignalperson** [comment]
With my surface pro 5 /w Version 1.1.0-hotfix+1 both the pen side and eraser side were acting as a marker.

I uninstalled a previous version of xournal++ before installing the latest, and remembered seeing a popup say something like "copied settings from .xournalpp to appdata/...".  Eventually I figured to try deleting the settings (under C:\Users\$username\AppData\Local\xournalpp) and happy to say that solved the problem.
